dev-db/clickhouse/clickhouse-9999.ebuild

```diff
@@ -0,0 +1,724 @@
+		cmd+='/add_library(unwind_static STATIC \${LIBUNWIND_SOURCES}/'
+
+		if use '!static' && use static-libs; then
+			cmd+=';s/target_compile_options(unwind_static PRIVATE'
+			cmd+='/target_compile_options(unwind_static PRIVATE -fPIC/'
+		fi
+
+		sed --in-place "${cmd}" contrib/libunwind-cmake/CMakeLists.txt || die "Failed to fixup libunwind"
+	fi
+
+	if use '!static' && use static-libs; then
+		for lib in Util Foundation; do
+			echo "target_compile_options(\${LIBNAME} PRIVATE -fPIC)" >> \
+				"contrib/poco/${lib}/CMakeLists.txt" || \
+				die "Failed to add PIC to ${lib}"
+		done
+	fi
+
+	[[ "${PV}" == "9999" ]] && mkdir --parents contrib/boost/boost
+
+	local cmake_config_checks line cmake_var use_flag
+
+	cmake_config_checks="${CLICKHOUSE_CMAKE_CONFIG_CHECKS[@]}"
+
+	cmake_config_checks+=(
+		"CMAKE_INTERPROCEDURAL_OPTIMIZATION $(usex lto)"
+		"USE_RAPIDJSON \
+			$(usex json "$(usex '!cpu_flags_x86_avx2' json '!cpu_flags_x86_avx2')" json)"
+		"USE_SIMDJSON \
+			$(usex json "$(usex cpu_flags_x86_avx2 json cpu_flags_x86_avx2)" json)"
+	)
+
+	for line in "${CLICKHOUSE_CMAKE_CONFIG_CHECKS[@]}"; do
+		read -r cmake_var use_flag <<< "${line}"
+		echo "if ($(
+			[[ "${use_flag}" ]] && usex "${use_flag}" "NOT" "" || echo NOT
+		) ${cmake_var})" >> CMakeLists.txt
+		echo "    message (FATAL_ERROR \"Failed to $(
+			[[ "${use_flag}" ]] && usex "${use_flag}" use disable || echo check
+		) ${use_flag}: ${cmake_var}\")" >> CMakeLists.txt
+		echo "endif ()" >> CMakeLists.txt
+	done
+}
+
+src_configure() {
+	use cpu_flags_x86_sse4_1 && append-cflags "-msse4.1" && append-cxxflags "-msse4.1"
+	use cpu_flags_x86_sse4_2 && append-cflags "-msse4.2" && append-cxxflags "-msse4.2"
+	use cpu_flags_x86_ssse3 && append-cflags "-mssse3" && append-cxxflags "-mssse3"
+	use cpu_flags_x86_avx && append-cflags "-mavx" && append-cxxflags "-mavx"
+	use cpu_flags_x86_avx2 && append-cflags "-mavx2" && append-cxxflags "-mavx2"
+	use cpu_flags_x86_popcnt && append-cflags "-mpopcnt" && append-cxxflags "-mpopcnt"
+
+	if use ppc64; then
+		 append-cflags "-maltivec" "-D__SSE2__=1" "-DNO_WARN_X86_INTRINSICS"
+		 append-cxxflags "-maltivec" "-D__SSE2__=1" "-DNO_WARN_X86_INTRINSICS"
+	fi
+
+	local line mycmakeargs use_flag cmake_option cmake_var
+	mycmakeargs=(
+		-D"CMAKE_INTERPROCEDURAL_OPTIMIZATION"="$(usex lto ON OFF)"
+		-D"DISABLE_CPU_OPTIMIZE"="ON"
+		-D"HAVE_SSE41"="$(usex cpu_flags_x86_sse4_1)"
+		-D"HAVE_SSE42"="$(usex cpu_flags_x86_sse4_2)"
+		-D"HAVE_SSSE3"="$(usex cpu_flags_x86_ssse3)"
+		-D"HAVE_AVX"="$(usex cpu_flags_x86_avx)"
+		-D"HAVE_AVX2"="$(usex cpu_flags_x86_avx2)"
+		-D"HAVE_POPCNT"="$(usex cpu_flags_x86_popcnt)"
+	)
+
+	for line in "${CLICKHOUSE_CMAKE_USE_ARGUMENTS[@]}"; do
+		read -r cmake_option use_flag <<< "${line}"
+		mycmakeargs+=( "-D${cmake_option}=$(
+			[[ ${use_flag} ]] && usex "${use_flag}" ON OFF || echo ON
+		)" )
+	done
+
+	mycmakeargs+=(
+		-D"BUILD_SHARED_LIBS"="$(usex static-libs OFF ON)"
+		-D"USE_STATIC_LIBRARIES"="$(usex static ON OFF)"
```

Comment:
Does this package really require ON/OFF? CMake is normally entirely happy with the default yes/no.

Reply:
Is it some kind of official style request? It is just the case that some options are going to be set automatically and ON/OFF is going to be used then. So CMakeCache.txt looks more consistent for me

Reply:
Now it is yes/no everywhere